detect_secrets/plugins/softlayer.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import
+
+import re
+
+import requests
+
+from .base import RegexBasedDetector
+from detect_secrets.core.constants import VerifiedResult
+
+
+class SoftlayerDetector(RegexBasedDetector):
+    """Scans for Softlayer credentials."""
+
+    secret_type = 'SoftLayer Credentials'
+
+    # opt means optional
+    sl = r'(?:softlayer|sl)(?:_|-|)(?:api|)'
+    key_or_pass = r'(?:key|pwd|password|pass|token)'
+    secret = r'([a-z0-9]{64})'
+    denylist = [
+        RegexBasedDetector.assign_regex_generator(
+            prefix_regex=sl,
+            password_keyword_regex=key_or_pass,
+            password_regex=secret,
+        ),
+
+        re.compile(
+            r'(?:http|https)://api.softlayer.com/soap/(?:v3|v3.1)/([a-z0-9]{64})',
+            flags=re.IGNORECASE,
+        ),
+    ]
+
+    def verify(self, token, content):
+        usernames = find_username(content)
+        if not usernames:
+            return VerifiedResult.UNVERIFIED
+
+        for username in usernames:
+            return verify_softlayer_key(username, token)
+
+        return VerifiedResult.VERIFIED_FALSE
+
+
+def find_username(content):
+    # opt means optional
+    username_keyword = r'(?:username|id|user|userid|user-id|user-name|' + \
+        r'name|user_id|user_name|uname)'
+    username = r'(\w(?:\w|_|@|\.|-)+)'
+    regex = re.compile(
+        RegexBasedDetector.assign_regex_generator(
+            prefix_regex=SoftlayerDetector.sl,
+            password_keyword_regex=username_keyword,
+            password_regex=username,
+        ),
+    )
+
+    return [
+        match
+        for line in content.splitlines()
+        for match in regex.findall(line)
+    ]
+
+
+def verify_softlayer_key(username, token):
+    try:
+        headers = {'Content-type': 'application/json'}
+        response = requests.get(
+            'https://api.softlayer.com/rest/v3/SoftLayer_Account.json',
+            auth=(username, token), headers=headers,
+        )
+
+        if response.status_code == 200:
+            return VerifiedResult.VERIFIED_TRUE
+        else:
+            return VerifiedResult.VERIFIED_FALSE
+    except Exception:
```

Comment:
> call can throw, or just requests.exceptions.RequestException. 

Good point.

> (Up to you if you want to try retrying on e.g. requests.exceptions.Timeout)

I will leave retry out.

> For posterity, this made me think about if we want to do this for other plugins like Slack Stripe etc

Wrapping some exception would be necessary, otherwise with current code, if Slack, Stripe or AWS verification endpoint is down or unreachable for whatever reason, we will throw exception instead of gracefully report unverified. As you said, it can force user to use `--no-verify` to keep using d-s.